### PR TITLE
Improve start time by adding index to e2e_cross_signing_keys

### PR DESCRIPTION
**Pull request by erikjohnston**
We do a `SELECT MAX(stream_id) FROM e2e_cross_signing_keys` on startup.

### CODE REVIEW

**Comment by clokep** [comment]
This seems to be failing with a `KeyError: 'e2e_cross_signing_keys_idx'` error?

**Comment by erikjohnston** [comment]
Fixed!

**Review by anoadragon453 — COMMENTED** [review]
Looks sane, though trying to understand why we register the index twice. For other indexes (e.g `user_ips_last_seen_only_index`, I see we only register once).

File: synapse/storage/databases/main/end_to_end_keys.py [hunk 1]
```diff
@@ -47,7 +48,20 @@ class DeviceKeyLookupResult:
...
+    def __init__(self, database: DatabasePool, db_conn: Connection, hs: "HomeServer"):
+        super().__init__(database, db_conn, hs)
+
+        self.db_pool.updates.register_background_index_update(
+            "e2e_cross_signing_keys_idx",
+            index_name="e2e_cross_signing_keys_stream_idx",
+            table="e2e_cross_signing_keys",
+            columns=["stream_id"],
+            unique=True,
+        )
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply synapse/storage/databases/main/end_to_end_keys.py:61]: I'm a crank and failed to commit the deletion of the one in `EndToEndKeyStore`...

**Review by anoadragon453 — APPROVED** [review]
lgtm!

**Pull request merged**